MoltenVK/MoltenVK/GPUObjects/MVKPixelFormats.mm

```diff
@@ -722,7 +722,7 @@
 	}
 
 #if MVK_XCODE_15
-	if (supportAtomics) {
+	if (supportAtomics && (mtlFormat == MTLPixelFormatR32Uint || mtlFormat == MTLPixelFormatR32Sint || mtlFormat == MTLPixelFormatRG32Uint)) {
```

Comment:
Does `supportAtomics` not include these additional checks already, via the setting of `MVKImage::_shouldSupportAtomics`?

Maybe add `VK_FORMAT_R32G32_UINT` to the `_shouldSupportAtomics` setting test, and also review and confirm the applicability of the additional tests for view-class there?

I'd like to avoid having same-but-not-the same logic in different places, unless there is some necessary reason why it's needed. And then document the difference if there is one.

Reply:
Not quite - this test fails the `_hasMutableFormat` case where pixel format is different, but the image support integer views. We could instead disable `_shouldSupportAtomics`. My rationale for putting it here was that it still *should* support atomics, we just can't set the usage bit. Like explained in the issue, that seems to work fine, but may not on another device or after an update, and if so we'll need to do something more intrusive.

Reply:
Okay. We'll see what happens with CTS testing.